lib/std/os/linux.zig

```diff
@@ -440,28 +440,51 @@ pub fn read(fd: i32, buf: [*]u8, count: usize) usize {
 }
 
 pub fn preadv(fd: i32, iov: [*]const iovec, count: usize, offset: i64) usize {
-    const offset_halves = splitValueLE64(offset);
-    return syscall5(
-        .preadv,
-        @bitCast(usize, @as(isize, fd)),
-        @ptrToInt(iov),
-        count,
-        offset_halves[0],
-        offset_halves[1],
-    );
+    if (usize_bits < 64) {
```

Comment:
It would better match the kernel logic if instead of branching here we just passed `@truncate(usize, @bitCast(u64, offset))` and `@truncate(usize, @bitCast(u64, offset) >> 32)`.